cmake_format/__main__.py

```diff
@@ -325,13 +357,34 @@ def main():
                           newline='')
 
     parse_ok = True
+    ignored = False
     if infile_path == '-':
       infile = io.open(os.dup(sys.stdin.fileno()),
                        mode='r', encoding='utf-8', newline='')
     else:
+      if not os.path.exists(infile_path):
+        sys.stderr.write(
+            "cmake-format error: File is not found: '{0}' \n".format(infile_path))
+        parse_ok = False
+        continue
       infile = io.open(infile_path, 'r', encoding='utf-8')
-
     try:
+
+      for ignore_dir in cfg.ignore_directories:
+        if ignore_dir in infile_path:
+          ignored = True
+          break
+
+      if ignored:
+        parse_ok = False
+        continue
+
+      if infile_path != '-' and not re.findall(CMAKE_LISTS_RE_PATTERNS, infile_path):
+        sys.stderr.write(
+            "cmake-format error: It is not cmake file: '{0}' \n".format(infile_path))
+        parse_ok = False
+        continue
+
```

Comment:
@dkandyba Please, pay attention to formatting of this for scope. "If" blocks are not alimented.

Reply:
@IGapchuk This is normal, that the "if" blocks are in that scope.

Reply:
@dkandyba Make alignment of all blocks inside loop